src/stateManager/validationStrategy/SpectatingValidationStrategy.ts

```diff
@@ -128,24 +181,48 @@ export default class SpectatingValidationStrategy extends AValidationStrategy {
         return BlockValidationResult.NOT_READY;
     }
     public async blockIsNotLinkedAndIsNotFirstBlock(
-        _block: Block
+        block: Block
     ): Promise<BlockValidationResult> {
+        const nextBlockHeight = this.storage.blocks.getNextBlockHeight(
+            block.forkId
+        );
+        if (nextBlockHeight === 0) {
+            this.storage.queues.queueBlock(block);
+            return BlockValidationResult.NOT_READY;
+        }
+
         this.disconnect();
         return BlockValidationResult.DISCONNECT;
     }
     public async objectiveInvalidTimestampDetected(
-        _block: Block
+        block: Block
     ): Promise<BlockValidationResult> {
+        if (!this.isSynced()) {
+            this.storage.queues.queueBlock(block);
+            return BlockValidationResult.NOT_READY;
+        }
+
         this.disconnect();
         return BlockValidationResult.DISPUTE;
     }
     public async subjectiveInvalidTimestampDetected(
-        _block: Block
+        block: Block
     ): Promise<BlockValidationResult> {
+        if (!this.isSynced()) {
+            this.storage.queues.queueBlock(block);
+            return BlockValidationResult.NOT_READY;
+        }
+
         return BlockValidationResult.NOT_ENOUGH_TIME;
     }
 
     private disconnect() {
         this.p2pManager.disconnectAll();
     }
+
+    private isSynced(): boolean {
+        const forkId = this.p2pManager.stateManager.forkId;
+        if (!forkId) return false;
+        return this.storage.blocks.getNextBlockHeight(forkId) > 0;
+    }
```

Comment:
You need to think more about this entire file - My quick though: future blocks should be queued under this strategy and maybe subjective failures. Any objective failure should result in disconnecting and aborting